sql: TestValidationWithProtectedTS failed

sql.TestValidationWithProtectedTS [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/11389897?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/11389897?buildTab=artifacts#/) on master @ [9ce61b0e79ca2c6ee219bc5a798fd3ff40d0367a](https://github.com/cockroachdb/cockroach/commits/9ce61b0e79ca2c6ee219bc5a798fd3ff40d0367a):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 52886 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestValidationWithProtectedTS
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/7786b2439f6fbd6878e73b0041b1a786/logTestValidationWithProtectedTS872886648
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:118: test server using tenant; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestValidationWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30752

--- a/pkg/sql/backfill_protected_timestamp_test.go
+++ b/pkg/sql/backfill_protected_timestamp_test.go
@@ -119,6 +119,7 @@ func TestValidationWithProtectedTS(t *testing.T) {
 		rSys.Exec(t, sql)
 	}
 	for _, sql := range []string{
+		"SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false",
 		"ALTER DATABASE defaultdb CONFIGURE ZONE USING gc.ttlseconds = 1",
 		"CREATE TABLE t(n int)",
 		"ALTER TABLE t CONFIGURE ZONE USING range_min_bytes = 0, range_max_bytes = 67108864, gc.ttlseconds = 1",

